packages/modules-sdk/src/medusa-module.ts

```diff
@@ -29,9 +29,18 @@ declare global {
   }
 }
 
+type ModuleAlias = {
+  key: string
+  hash: string
+  alias?: string
+  main?: boolean
+}
+
 export class MedusaModule {
   private static instances_: Map<string, any> = new Map()
+  private static keys_: Map<string, ModuleAlias[]> = new Map()
```

Comment:
todo: if we could set a module meaningful name :) something like `modules`?

Reply:
We also need to change the way a module is registered in the container, we might need to use the alias the concatenate with the registration name? or something like that to avoid overriding a registration with another

Reply:
yeah agree, let's wait but at least we have that in mind